### PR TITLE
Add ability to generate a quote to `mc-sgx-dcap-ql`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #147** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/147" target="_blank">Graphite</a> 
    * **PR #148** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/148" target="_blank">Graphite</a> 
      * **PR #149** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/149" target="_blank">Graphite</a> 
        * **PR #150** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/150" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/150?utm_source=stack-comment).

File: dcap/ql/src/quote_generator.rs
Comment (nick-mobilecoin) [inline dcap/ql/src/quote_generator.rs:39]: I'm thinking we'll want to create a `Quote3` type longer term

File: dcap/ql/src/quote_generator.rs
Comment (nick-mobilecoin) [inline dcap/ql/src/quote_generator.rs:23]: Curious about thoughts here? It felt like the simplest way to break the psuedo circular dependency since we need the target_info from the QE, to create a report, but then we need the report for the PCE to generate a quote.   
I'm not sure we should expose the QE vs PCE interfaces to the users as separate structs/traits

File: dcap/ql/src/quote_generator.rs
Comment (nick-mobilecoin) [inline dcap/ql/src/quote_generator.rs:41]: I'm up for ideas on testing, I can continue to build the HW tests.  I don't like that we currently only have a manual way to run them right now.
I could also look at out of process quoting and maybe mocking those interfaces out for tests

**Review by NotGyro — APPROVED** [review]
This all looks good to me.

File: dcap/ql/src/quote_generator.rs
Comment (jcape) [reply dcap/ql/src/quote_generator.rs:23]: I think this interface should be inverted to traits monkey-patched onto the structs in question. Like, `TargetInfo::for_quoting_enclave()`

File: dcap/ql/src/quote_generator.rs
Comment (jcape) [reply dcap/ql/src/quote_generator.rs:39]: Similarly, this should be, e.g. `TryFrom<Report> for Quote3`

File: dcap/ql/src/quote_generator.rs
Comment (nick-mobilecoin) [reply dcap/ql/src/quote_generator.rs]: code changed, comment no longer applies

File: dcap/ql/src/quote_generator.rs
Comment (nick-mobilecoin) [reply dcap/ql/src/quote_generator.rs]: code changed so comment out of context.

File: dcap/ql/src/quote3.rs
Comment (nick-mobilecoin) [inline dcap/ql/src/quote3.rs:18]: @jcape I know you mentioned [u8] being sufficient for the RawQuote/BaseQuote, wasn't sure if you thought so here too.  If so are you thinking this returns a Vec<u8>.  Due to the sizing issue I'm not fond of the caller having to pass in a pre-sized `[u8]`

**Review by jcape — APPROVED** [review]

**Review by NotGyro — APPROVED** [review]
As stated previously, looks good to me - the TryFrom implementation is a nice improvement.

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/150?utm_source=stack-comment).


Job Reference: MTlCz31pHORkTM9rocyb

**Pull request merged**
